src/Microsoft.ML.Dnn/ImageClassificationTrainer.cs

```diff
@@ -11,1598 +7,1456 @@
-
-            tf_with(tf.name_scope("accuracy"), scope =>
+            /// <summary>
+            /// Indicates the kind of the dataset of which metric is reported.
+            /// </summary>
+            public enum Dataset
             {
-                tf_with(tf.name_scope("correct_prediction"), delegate
-                {
-                    _prediction = tf.argmax(resultTensor, 1);
-                    correctPrediction = tf.equal(_prediction, groundTruthTensor);
-                });
+                Train,
+                Validation
+            };
 
-                tf_with(tf.name_scope("accuracy"), delegate
-                {
-                    evaluationStep = tf.reduce_mean(tf.cast(correctPrediction, tf.float32));
-                });
-            });
+            /// <summary>
+            /// Contains train time metrics.
+            /// </summary>
+            public TrainMetrics Train { get; set; }
 
-            tf.summary.scalar("accuracy", evaluationStep);
-            return (evaluationStep, _prediction);
+            /// <summary>
+            /// Contains pre-train time metrics. These contains metrics on image
+            /// featurization.
+            /// </summary>
+            public BottleneckMetrics Bottleneck { get; set; }
+
+            /// <summary>
+            /// String representation of the metrics.
+            /// </summary>
+            public override string ToString() => Train != null ? Train.ToString() : Bottleneck.ToString();
         }
 
-        private void UpdateTransferLearningModelOnDisk(ImageClassificationEstimator.Options options, int classCount)
+        public sealed class Options : TrainerInputBaseWithLabel
         {
-            var (sess, _, _, _) = BuildEvaluationSession(options, classCount);
-            var graph = sess.graph;
-            var outputGraphDef = tf.graph_util.convert_variables_to_constants(
-                sess, graph.as_graph_def(), new string[] { _softMaxTensor.name.Split(':')[0], _prediction.name.Split(':')[0], _jpegData.name.Split(':')[0], _resizedImage.name.Split(':')[0] });
+            /// <summary>
+            /// Number of samples to use for mini-batch training.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Number of samples to use for mini-batch training.", SortOrder = 9)]
+            public int BatchSize = 64;
 
-            string frozenModelPath = _checkpointPath + ".pb";
-            File.WriteAllBytes(_checkpointPath + ".pb", outputGraphDef.ToByteArray());
-            _session.graph.Dispose();
-            _session.Dispose();
-            _session = LoadTFSessionByModelFilePath(_env, frozenModelPath, false);
-        }
+            /// <summary>
+            /// Number of training iterations.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Number of training iterations.", SortOrder = 10)]
+            public int Epoch = 100;
```

Comment:
You may want to align  with naming conventions of ML.NET instead of TF. This is called `numberOfIterations` in other ML.NET trainers.